lib/torrent.js

```diff
@@ -281,83 +278,78 @@ class Torrent extends EventEmitter {
+            extendMutable(opts, this.client.tracker.getAnnounceOpts())
           }
-          if (self._getAnnounceOpts) {
+          if (this._getAnnounceOpts) {
             // TODO: consider deprecating this, as it's redundant with the former case
-            extendMutable(opts, self._getAnnounceOpts())
+            extendMutable(opts, this._getAnnounceOpts())
           }
           return opts
         }
       })
     }
 
     // begin discovering peers via DHT and trackers
-    self.discovery = new Discovery({
-      infoHash: self.infoHash,
-      announce: self.announce,
-      peerId: self.client.peerId,
-      dht: !self.private && self.client.dht,
+    this.discovery = new Discovery({
+      infoHash: this.infoHash,
+      announce: this.announce,
+      peerId: this.client.peerId,
+      dht: !this.private && this.client.dht,
       tracker: trackerOpts,
-      port: self.client.torrentPort,
+      port: this.client.torrentPort,
       userAgent: USER_AGENT
     })
 
-    self.discovery.on('error', onError)
-    self.discovery.on('peer', onPeer)
-    self.discovery.on('trackerAnnounce', onTrackerAnnounce)
-    self.discovery.on('dhtAnnounce', onDHTAnnounce)
-    self.discovery.on('warning', onWarning)
-
-    function onError (err) {
-      self._destroy(err)
-    }
+    this.discovery.on('error', (err) => {
+      this._destroy(err)
+    })
 
-    function onPeer (peer) {
+    this.discovery.on('peer', (peer) => {
       // Don't create new outgoing TCP connections when torrent is done
-      if (typeof peer === 'string' && self.done) return
-      self.addPeer(peer)
-    }
+      if (typeof peer === 'string' && this.done) return
+      this.addPeer(peer)
+    })
 
-    function onTrackerAnnounce () {
-      self.emit('trackerAnnounce')
-      if (self.numPeers === 0) self.emit('noPeers', 'tracker')
-    }
+    this.discovery.on('trackerAnnounce', () => {
+      this.emit('trackerAnnounce')
+      if (this.numPeers === 0) this.emit('noPeers', 'tracker')
+    })
 
-    function onDHTAnnounce () {
-      self.emit('dhtAnnounce')
-      if (self.numPeers === 0) self.emit('noPeers', 'dht')
-    }
+    this.discovery.on('dhtAnnounce', () => {
+      this.emit('dhtAnnounce')
+      if (this.numPeers === 0) this.emit('noPeers', 'dht')
+    })
 
-    function onWarning (err) {
-      self.emit('warning', err)
-    }
+    this.discovery.on('warning', (err) => {
+      this.emit('warning', err)
+    })
 
-    if (self.info) {
+    if (this.info) {
       // if full metadata was included in initial torrent id, use it immediately. Otherwise,
       // wait for torrent-discovery to find peers and ut_metadata to get the metadata.
-      self._onMetadata(self)
-    } else if (self.xs) {
-      self._getMetadataFromServer()
+      this._onMetadata(this)
+    } else if (this.xs) {
+      this._getMetadataFromServer()
     }
   }
 
   _getMetadataFromServer () {
+    // to allow function hoisting
```

Comment:
Hoisting was used here for readability, and hoisting is impossible with arrow functions